src/platform_impl/linux/x11/mod.rs

```diff
@@ -301,10 +309,7 @@ impl<T: 'static> EventLoop<T> {
             }
             // Empty the redraw requests
             {
-                // Release the lock to prevent deadlock
-                let windows: Vec<_> = wt.pending_redraws.lock().unwrap().drain().collect();
-
-                for wid in windows {
+                while let Ok(wid) = self.redraw_channel.try_recv() {
```

Comment:
Can we name this `window_id`? I feel like we don't gain anything by using a cryptic `wid` identifier here.